### PR TITLE
www-apps/webdavcgi: Version-Bump to 1.1.2 and cleanup

### PR DESCRIPTION
**Pull request by paraenggu**
The following PR bumps `www-apps/webdavcgi` to the [latest upstream 1.1.2 version](https://github.com/DanRohde/webdavcgi/releases/tag/1.1.2).

It also addresses [#658470](https://bugs.gentoo.org/658470), [#674772](https://bugs.gentoo.org/674772), [#644828](https://bugs.gentoo.org/644828) and cleans-up old and obsolete ebuilds.

### CODE REVIEW

**Comment by paraenggu** [comment]
Thanks for testing!

**Comment by paraenggu** [comment]
@gentoo/proxy-maint is there something missing from my side are there any additional steps required to proceed? Thanks for your feedback!

**Comment by kentfredric** [comment]
I'll do the dev-perl/IO-Compress-Brotli myself and merge it today so that your PR won't have to worry about it, and its really the domain of perl@ team to maintain this.

If you want to be a co-maintainer still, that's fine.

Just as-is there's a handful of defects vs perl standards that have to be ironed out.

**Comment by paraenggu** [comment]
Hi kentfredric 

> I'll do the dev-perl/IO-Compress-Brotli myself and merge it today so that your PR won't have to worry about it, and its really the domain of perl@ team to maintain this.

I dropped the `dev-perl/IO-Compress-Brotli` ebuild from this PR.

**Comment by kentfredric** [comment]
> Hi @kentfredric
> 
> > I'll do the dev-perl/IO-Compress-Brotli myself and merge it today so that your PR won't have to worry about it, and its really the domain of perl@ team to maintain this.
> 
> I dropped the `dev-perl/IO-Compress-Brotli` ebuild from this PR.
> 
> > Just as-is there's a handful of defects vs perl standards that have to be ironed out.
> 
> Please let me know what has to be changed or apply the changes directly (I've allowd "edits from maintainers" on this PR) - thanks.

The impetus wasn't personal, just packaging perl stuff from scratch is less work than reviewing it. ( Generally, in order to simply review it to be correct, you basically have to re-implement it yourself in the process ).

And also the hope is to reduce the scope of this PR, so that *some* of it can be done without contention.

Webdav and friends aren't something I have the patience to review or merge, but I *can* do the perl part and make it so nobody else has to think about it, which, will hopefully accelerate the merging of the rest of this PR.

Yes, generally I'd want feedback too. Just packaging dev-perl/ stuff basically requires to to know a lot about the micro-ecosystem which has a lot of extra faff that is not standard for people to know for general ebuild work.

Just in future, if you need a new dev-perl/ package, don't hurt yourself trying to package it, the easiest way to get it in quickly is to file a bug on bugs.gentoo.org requesting it, mark it as a blocker for whatever you're doing, and let perl@ herd look after it. ( Generally, we keep the number of packages in `dev-perl/` not maintained by perl@ to an absolute minimum, because the absence of `perl@` generally causes it to be _less_ maintained, not more )

File: www-apps/webdavcgi/webdavcgi-1.1.2.ebuild [hunk 1]
```diff
@@ -162,23 +173,34 @@ pkg_postinst() {
 	elog "The WebDAV CGI config is located at ${WEBDAVCONFIG}."
 	elog
 	elog "An example Apache HTTP server configuration snippet is available in"
-	elog "${ROOT%/}/usr/share/doc/${PF} in the file ${APACHEEXAMPLECONFIG}"
+	elog "${ROOT}/usr/share/doc/${PF} in the file ${APACHEEXAMPLECONFIG}"
```
Comment (thesamesam) [inline www-apps/webdavcgi/webdavcgi-1.1.2.ebuild:176]: ${EROOT}?

File: www-apps/webdavcgi/webdavcgi-1.1.2.ebuild [hunk 2]
```diff
@@ -55,6 +56,16 @@ REQUIRED_USE="|| ( mysql postgres sqlite )"
...
+	# The patch originates from
+	# https://github.com/DanRohde/webdavcgi/commit/04e79b7ecbaf3aae5ab813cd4fc0a009c72b1580
+	# and can be remove as soon as this ebuild gets bumped to 1.1.3 which
+	# already includes the change.
+	"${FILESDIR}/${P}-fix-unescaped-braces.patch"
+)
+
 src_compile() {
 	if use suid; then
 		WEBDAVWRAPPERS="webdavwrapper" # Standard UID/GID wrapper
```
Comment (thesamesam) [inline www-apps/webdavcgi/webdavcgi-1.1.2.ebuild:71]: Is the export definitely needed?

File: www-apps/webdavcgi/webdavcgi-1.1.2.ebuild [hunk 3]
```diff
@@ -1,20 +1,20 @@
-# Copyright 1999-2016 Gentoo Foundation
+# Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
 
-inherit eutils toolchain-funcs versionator
+inherit eutils toolchain-funcs
```
Comment (thesamesam) [inline www-apps/webdavcgi/webdavcgi-1.1.2.ebuild:6]: Are we using eutils?

**Review by thesamesam — COMMENTED** [review]
Minor comments. Again, I'm sorry this was forgot about, but it is on my list now.

**Comment by thesamesam** [comment]
@paraenggu ping

**Comment by juippis** [comment]
Ping.

File: www-apps/webdavcgi/webdavcgi-1.1.2.ebuild (hunk 1 above)
Comment (paraenggu) [reply www-apps/webdavcgi/webdavcgi-1.1.2.ebuild]: Switched to `${EROOT}`, thanks.

File: www-apps/webdavcgi/webdavcgi-1.1.2.ebuild (hunk 2 above)
Comment (paraenggu) [reply www-apps/webdavcgi/webdavcgi-1.1.2.ebuild:71]: `WEBDAVWRAPPERS` is used within `src_install()` later on.

File: www-apps/webdavcgi/webdavcgi-1.1.2.ebuild (hunk 3 above)
Comment (paraenggu) [reply www-apps/webdavcgi/webdavcgi-1.1.2.ebuild]: No, thanks for spotting, `eutils` was removed form `inherit`.

**Comment by paraenggu** [comment]
Sorry for the delay, I've been too busy lately. I finally found some time to address the issues reported by @thesamesam and rebased everything to the current master.

**Review by juippis — APPROVED** [review]
I can't do runtime testing, but looks good to me.

**Comment by paraenggu** [comment]
@thesamesam / @juippis Thanks a lot for your review, approval and merging!

**Pull request closed without merging**